ui/js/perf.js

```diff
@@ -659,3 +659,157 @@ perf.filter('displayPrecision', function() {
         return parseFloat(input).toFixed(2);
     };
 });
+
+
+perf.service('JsonPushes', ['$http','$q', function($http, $q) {
```

Comment:
Should be a space between '$http' and '$q'. I might also make this a treeherder service since it might also be useful in treeherder itself, could you create a new file in `ui/js/services/jsonpushes.js` and hang this off of treeherder (it should still be usable by perfherder).